commit 994a03cdee262e533d42422084f5066bc8cc7cbf
Author: Jonathan Ellis <jbellis@apache.org>
Date:   Wed Aug 31 18:17:06 2011 +0000

    add delay parameter to removeToken so RemoveTest doesn't have to wait for full RING_DELAY
    patch by jbellis
    
    git-svn-id: https://svn.apache.org/repos/asf/cassandra/trunk@1163731 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/src/java/org/apache/cassandra/gms/Gossiper.java b/src/java/org/apache/cassandra/gms/Gossiper.java
index e6dc2a4..79fdcae 100644
--- a/src/java/org/apache/cassandra/gms/Gossiper.java
+++ b/src/java/org/apache/cassandra/gms/Gossiper.java
@@ -362,8 +362,9 @@ public class Gossiper implements IFailureDetectionEventListener, GossiperMBean
      * @param endpoint - the endpoint being removed
      * @param token - the token being removed
      * @param mytoken - my own token for replication coordination
+     * @param delay
      */
-    public void advertiseRemoving(InetAddress endpoint, Token token, Token mytoken)
+    public void advertiseRemoving(InetAddress endpoint, Token token, Token mytoken, int delay)
     {
         EndpointState epState = endpointStateMap.get(endpoint);
         // remember this node's generation
@@ -372,7 +373,7 @@ public class Gossiper implements IFailureDetectionEventListener, GossiperMBean
         logger.info("Sleeping for " + StorageService.RING_DELAY + "ms to ensure " + endpoint + " does not change");
         try
         {
-            Thread.sleep(StorageService.RING_DELAY);
+            Thread.sleep(delay);
         }
         catch (InterruptedException e)
         {
diff --git a/src/java/org/apache/cassandra/service/StorageService.java b/src/java/org/apache/cassandra/service/StorageService.java
index 2be322b..b93c9aa 100644
--- a/src/java/org/apache/cassandra/service/StorageService.java
+++ b/src/java/org/apache/cassandra/service/StorageService.java
@@ -27,6 +27,7 @@ import java.net.UnknownHostException;
 import java.nio.ByteBuffer;
 import java.util.*;
 import java.util.concurrent.*;
+import javax.lang.model.type.TypeKind;
 import javax.management.MBeanServer;
 import javax.management.ObjectName;
 
@@ -2107,6 +2108,11 @@ public class StorageService implements IEndpointStateChangeSubscriber, StorageSe
      */
     public void removeToken(String tokenString)
     {
+        removeToken(tokenString, RING_DELAY);
+    }
+
+    public void removeToken(String tokenString, int delay)
+    {
         InetAddress myAddress = FBUtilities.getBroadcastAddress();
         Token localToken = tokenMetadata_.getToken(myAddress);
         Token token = partitioner.getTokenFactory().fromString(tokenString);
@@ -2153,7 +2159,7 @@ public class StorageService implements IEndpointStateChangeSubscriber, StorageSe
         calculatePendingRanges();
         // the gossiper will handle spoofing this node's state to REMOVING_TOKEN for us
         // we add our own token so other nodes to let us know when they're done
-        Gossiper.instance.advertiseRemoving(endpoint, token, localToken);
+        Gossiper.instance.advertiseRemoving(endpoint, token, localToken, delay);
 
         // kick off streaming commands
         restoreReplicaCount(endpoint, myAddress);
diff --git a/test/unit/org/apache/cassandra/service/RemoveTest.java b/test/unit/org/apache/cassandra/service/RemoveTest.java
index e0b0bee..0a7a325 100644
--- a/test/unit/org/apache/cassandra/service/RemoveTest.java
+++ b/test/unit/org/apache/cassandra/service/RemoveTest.java
@@ -124,7 +124,7 @@ public class RemoveTest extends CleanupHelper
             {
                 try
                 {
-                    ss.removeToken(token);
+                    ss.removeToken(token, 0);
                 }
                 catch (Exception e)
                 {
